commitserver/commit/hydratorhelper.go

```diff
@@ -204,6 +204,17 @@ func writeManifests(root *os.Root, dirPath string, manifests []*apiclient.Hydrat
 		if err != nil {
 			return fmt.Errorf("failed to unmarshal manifest: %w", err)
 		}
+		// Remove Argo CD's internal tracking annotation before committing to Git.
+		// This annotation is used only for in-memory tracking and should not be persisted.
+		annotations := obj.GetAnnotations()
+		if annotations != nil {
+			delete(annotations, common.AnnotationKeyAppInstance)
+			if len(annotations) == 0 {
+				obj.SetAnnotations(nil)
+			} else {
+				obj.SetAnnotations(annotations)
+			}
+		}
```

Comment:
Instead of adding the behavior inline here, how about adding a new `RemoveAppInstance` method to the `argo.ResourceTracking` interface? Some of the logic from `SetAppInstance` could be duplicated to cover both the label and annotation cases.

Reply:
I researched a bit more about the flow of code and I want to ask something here:

Reading the following code here:

1. https://github.com/argoproj/argo-cd/blob/master/controller/hydrator/hydrator.go#L311
2. https://github.com/argoproj/argo-cd/blob/master/controller/state.go#L304
3. https://github.com/argoproj/argo-cd/blob/master/reposerver/repository/repository.go#L1562 (this is where app instance is set)

Even after adding the `RemoveAppInstance` method to the interface, Is it idle to call that method for removing the app instance fields once the app instance is set in `repository.go`?  What would you suggest here doing?

Reply:
Let's say we go with the option 4. Even if we add that, we are still required to call that function before committing the manifest. Isn't it? So, how is adding the method benefiting us when we are still doing the inline addition (calling that function) before committing?

Reply:
For the reasons in this comment: https://github.com/argoproj/argo-cd/pull/23857#discussion_r2219542292

Reply:
That's alright. If you look at the changes now, the `RemoveAppInstance` is literally a dead code. Hence, if we were to manually range over objects and remove it, what's the use of the function we wrote instead?